File: java/dtdgen-master/src/main/java/com/philhanna/dtdgen/modelbuilder/StackEntry.java
package com.philhanna.dtdgen.modelbuilder;

import com.philhanna.dtdgen.ElementModel;

/**
 * StackEntry is a data structure we put on the stack for each nested
 * element
 */
public class StackEntry {

   // ====================================================================
   // Class constants and variables
   // ====================================================================

   // ====================================================================
   // Class methods
   // ====================================================================

   // ====================================================================
   // Instance variables
   // ====================================================================

   private final ElementModel elementModel;
   private int sequenceNumber = -1;
   private String latestChildName;

   // ====================================================================
   // Constructors
   // ====================================================================

   public StackEntry(ElementModel elementModel) {
      this.elementModel = elementModel;
   }

   // ====================================================================
   // Instance methods
   // ====================================================================

   public ElementModel getElementModel() {
      return elementModel;
   }

   public int getSequenceNumber() {
      return sequenceNumber;
   }

   public void incrementSequenceNumber() {
      sequenceNumber++;
   }

   public String getLatestChildName() {
      return latestChildName;
   }

   public void setLatestChildName(String latestChild) {
      this.latestChildName = latestChild;
   }
}

File: dtdgen/model_builder/stack_entry.py
from typing import Optional

from dtdgen import ElementModel


class StackEntry:
    """A data structure that is put on the stack for each nested element"""
    def __init__(self, element_model: ElementModel):
        self._element_model: ElementModel = element_model
        self._sequence_number: int = -1
        self._latest_child_name: Optional[str] = None

    @property
    def element_model(self):
        return self._element_model

    @property
    def sequence_number(self):
        return self._sequence_number

    def increment_sequence_number(self):
        self._sequence_number += 1

    @property
    def latest_child_name(self):
        return

    @latest_child_name.setter
    def latest_child_name(self, value: str):
        self._latest_child_name = value
